net: sockets: getaddrinfo: increase sys heap via Kconfig

### Summary

`getaddrinfo` now uses `k_` variants for memory allocation and therefore needs system heap.
If something uses `getaddrinfo` but otherwise does not use the system heap and therefore system heap size might be 0 compilation fails.

### Describe the solution you'd like

Add Kconfig value `HEAP_MEM_POOL_ADD_SIZE_GETADDRINFO` to `subsys/net/lib/sockets/Kconfig`, e.g.:
```
config HEAP_MEM_POOL_ADD_SIZE_GETADDRINFO
        int
        default 512
```

### Alternatives

_No response_

### Additional Context

Reproduce with:
```sh
west build -p always -b native_sim samples/subsys/shell/shell_module/ -- -DCONFIG_NETWORKING=y -DCONFIG_NET_TCP=y -DCONFIG_NET_IPV4=y -DCONFIG_SHELL_BACKEND_MQTT=y
```

Compile error:
```sh
in function `zsock_getaddrinfo':
getaddrinfo.c:443:(.text.zsock_getaddrinfo+0xe): undefined reference to `k_calloc'
```

--- a/subsys/net/lib/sockets/Kconfig
+++ b/subsys/net/lib/sockets/Kconfig
@@ -69,6 +69,13 @@ config NET_SOCKETS_DNS_BACKOFF_INTERVAL
 	     If no reply is received, a 3rd query is done after 15 sec (5 + 5 * 2),
 	     and the timeout is set to 2 sec so that the total timeout is 17 seconds.
 
+config HEAP_MEM_POOL_ADD_SIZE_GETADDRINFO
+	# Defaults to heap memory needed for a single getaddrinfo() call in
+	# a default configuration on 64-bit platform
+	int
+	default 280
+	depends on DNS_RESOLVER
+
 config NET_SOCKET_MAX_SEND_WAIT
 	int "Max time in milliseconds waiting for a send command"
 	default 10000

